cmd/state/internal/cmdtree/intercepts/cmdcall/cmdcall.go

```diff
@@ -31,6 +32,15 @@ func (c *CmdCall) InterceptExec(next captain.ExecuteFunc) captain.ExecuteFunc {
 		}
 
 		if err := next(cmd, args); err != nil {
+			// check can be removed when no runners return failures and,
+			// possibly, when main pkg error handling digs through the error
+			// chain to get to nested failures
+			if fail, ok := err.(*failures.Failure); ok {
+				if fail == nil {
+					return nil
+				}
+				return fail // wrapping would break error handling higher in stack
```

Comment:
Why? The code in `main` know how to handle failures.

Reply:
The main error handling can handle a failure passed directly back to it, but it does not handle nested failures. I was correct in my suspicion that failures could implement locale.ErrorLocalizer and things would be eased.

The "ToError" calls I added are the shortest forms of ensuring a typed-nil is not propagated up the stack.

The auth client methods had some knock-on effects, but I capped them. The similar package-level function still uses the old functionality for now.

Reply:
The use of FailureType.Wrap() makes it too awkward to treat failures much like errors, or even equals in satisfying our own interfaces. This means I'm stuck at the main pkg error handling not handling nested failures correctly (especially with multiple failures in a chain).

Reply:
> but it does not handle nested failures

This is not true. It does handle nested failures. If you are running into a case where it doesn't then that's a bug and I'd like to hear more about it.

Reply:
FTR; Communicated in Slack with nothing to report for now.